src/Toml/BiMap.hs

```diff
@@ -96,41 +94,52 @@ _Right = invert $ prism (either (const Nothing) Just) Right
 -- | Creates prism for 'AnyValue'.
 mkAnyValueBiMap :: (forall t . Value t -> Maybe a)
                 -> (a -> Value tag)
-                -> BiMap AnyValue a
+                -> BiMap a AnyValue
 mkAnyValueBiMap matchValue toValue =
-    prism (\(AnyValue value) -> matchValue value) (AnyValue . toValue)
+    BiMap (Just . AnyValue . toValue) (\(AnyValue value) -> matchValue value)
+
+-- | Creates prism for 'Text' to 'AnyValue' with custom functions
+_TextBy :: (Text -> Maybe a) -> (a -> Text) -> BiMap a AnyValue
+_TextBy parseText toText =
+  mkAnyValueBiMap (matchText >=> parseText) (Text . toText)
 
 -- | Allows to match against given 'Value' using provided prism for 'AnyValue'.
-matchValueForward :: BiMap AnyValue a -> Value t -> Maybe a
-matchValueForward = liftMatch . forward
+matchValueBackward :: BiMap a AnyValue -> Value t -> Maybe a
+matchValueBackward = liftMatch . backward
 
 -- | 'Bool' bimap for 'AnyValue'. Usually used with 'arrayOf' combinator.
-_Bool :: BiMap AnyValue Bool
+_Bool :: BiMap Bool AnyValue
 _Bool = mkAnyValueBiMap matchBool Bool
 
 -- | 'Integer' bimap for 'AnyValue'. Usually used with 'arrayOf' combinator.
-_Integer :: BiMap AnyValue Integer
+_Integer :: BiMap Integer AnyValue
 _Integer = mkAnyValueBiMap matchInteger Integer
 
 -- | 'Double' bimap for 'AnyValue'. Usually used with 'arrayOf' combinator.
-_Double :: BiMap AnyValue Double
+_Double :: BiMap Double AnyValue
 _Double = mkAnyValueBiMap matchDouble Double
 
 -- | 'Text' bimap for 'AnyValue'. Usually used with 'arrayOf' combinator.
-_Text :: BiMap AnyValue Text
+_Text :: BiMap Text AnyValue
 _Text = mkAnyValueBiMap matchText Text
 
-_TextToString :: BiMap Text String
-_TextToString = iso T.unpack T.pack
+_StringToText :: BiMap String Text
+_StringToText = iso T.pack T.unpack
```

Comment:
I have better idea regarding naming convention: let's remove `To` part from such prism. So it will be just: `_StringText`. It doesn't make much sense to me now to have `To` part because conversion is bidirectional.